src/main/java/it/eng/idsa/dataapp/service/impl/ProxyServiceImpl.java

```diff
@@ -250,8 +255,10 @@ private String saveFileToDisk(String responseMessage, Message requestMessage) th
 			String reqArtifact = ((ArtifactRequestMessage) requestMessage).getRequestedArtifact().getPath();
 			// get resource from URI http://w3id.org/engrd/connector/artifact/ + requestedArtifact
 			requestedArtifact = reqArtifact.substring(reqArtifact.lastIndexOf('/') + 1);
-			logger.info("About to save file " + requestedArtifact);
-			recreateFileService.recreateTheFile(payload, new File(requestedArtifact));
+			String dataLake= dataLakeDirectory + File.pathSeparator + requestedArtifact;
+			logger.info("About to save file " + dataLake);
+			
+			recreateFileService.recreateTheFile(payload, new File(dataLakeDirectory + FileSystems.getDefault().getSeparator() + requestedArtifact));
```

Comment:
Why not reusing already created dataLake variable to create new File?
+ please apply formatting (for example, space after = 
and also in test file, after ,